### PR TITLE
Add email notifications for processing jobs

### LINKED ISSUE #6673: Notify users by email after a long-running jobs finishes
## Detailed Description
Right now, it is hard to tell when a long-running job finishes. Some users were not aware of the `Jobs` page altogether. I suggest to send users an email after each job finishes either with a link to the jobs page or even better directly with an actionable link, e.g. "View Dataset", "Download Tiffs" etc. 

Considerations up for discussion:
- Should we only send emails in the success case? Or also for failed jobs? 
- Should we blacklist some long-running jobs that do not produce a link, e.g. "find largest segment id"?

## Context

* [x] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
* [ ] Specific to webKnossos.org (set `isDemoInstance=true` in `application.conf`)

Long-running jobs are not easily discoverable when triggered somewhere in the UI. It also not easy to gauge the runtime of certain jobs. Notifying users of success - and perhaps equally important - of failures seems a good communication strategy.

### PR DESCRIPTION
**Pull request by hotzenklotz**
This PR adds email notifications for long running WEBKNOSSOS worker jobs. The PR also adds a new API route for downloading Tiffs that does not require a user token (but handles authentication internally). This URI can easily be shared through email without "exposing" a user token.

Jobs with email notifications:
- Compute meshes
- Upload dataset / Convert
- Export as Tiff images
- Globalize Flood-Fill
- Neuron Segmentation
- Nuclei Segmentation
- Materialize Volume Annotation

Jobs without email notifications:
- Find largest segment ID

[Design Doc](https://www.notion.so/scalableminds/Design-Doc-Email-Notifications-31a7b708e9a34bfaa0810c5c8cf49af1?pvs=4)

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
1. Prepare your WK instance for email sending in the application.conf
2. Enable jobs and run a worker
3. Execute a worker job, e.g. "Tiff Download"
4. Receive an email. 

### Issues:
- fixes #6673

~~### Requires WK worker update on deployment. See accompanying worker PR https://github.com/scalableminds/voxelytics/pull/3245~~
------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: app/views/mail/jobSuccessfulGeneric.scala.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@(name: String, datasetName: String, jobLink: String, jobTitle: String, jobDescription: String)
+
+@emailBaseTemplate() {
+<p>Hi @{name},</p>
+
+<p>We wanted to let you know that WEBKNOSSOS has finished your background job for the dataset <i>@{datasetName}</i>.
```
Comment (philippotto) [inline app/views/mail/jobSuccessfulGeneric.scala.html:6]: "We wanted to let you know that" could be removed in my opinion as it doesn't add much value.

File: app/views/mail/jobFailedUploadConvert.scala.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@(name: String, datasetName: String)
+
+@emailBaseTemplate() {
+<p>Hi @{name},</p>
+
+<p>
+  oops, something went unexpected wrong when uploading and converting your dataset <i>@{datasetName}</i> for WEBKNOSSOS.
```
Comment (philippotto) [inline app/views/mail/jobFailedUploadConvert.scala.html:7]: Maybe something like "unfortunately, your uploaded dataset could not be converted automatically for WEBKNOSSOS" would sound a bit more professional?

File: app/views/mail/jobFailedGeneric.scala.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+@(name: String, datasetName: String, jobTitle: String)
+
+@emailBaseTemplate() {
+<p>Hi @{name},</p>
+
+<p>
+  oops, something unexpected went wrong when working on your WEBKNOSSOS @{jobTitle} job for the <i>@{datasetName}</i> dataset.
```
Comment (philippotto) [inline app/views/mail/jobFailedGeneric.scala.html:7]:
```suggestion
  oops, something unexpected went wrong when executing your WEBKNOSSOS @{jobTitle} job for the <i>@{datasetName}</i> dataset.
```

**Review by philippotto — COMMENTED** [review]
Frontend looks good :+1: I left some feedback on wording :writing_hand:

File: app/models/job/JobCommand.scala [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  type JobCommand = Value
+
+  val COMPUTE_MESH_FILE = Value("compute_mesh_file")
+  val CONVERT_TO_WKW = Value("convert_to_wkw")
+  val EXPORT_TIFF = Value("export_tiff")
+  val FIND_LARGEST_SEGMENT_ID = Value("find_largest_segment_id")
+  val GLOBALIZE_FLOODFILLS = Value("globalize_floodfills")
+  val INFER_NUCLEI = Value("infer_nuclei")
+  val INFER_NEURONS = Value("infer_neurons")
+  val MATERIALIZE_VOLUME_ANNOTATION = Value("materialize_volume_annotation")
```
Comment (fm3) [inline app/models/job/JobCommand.scala:15]:
```suggestion
val compute_mesh_file, convert_to_wkw, export_tiff, find_largest_segment_id, globalize_floodfills, infer_nuclei, infer_neurons, materialize_volume_annotation = Value
```

**Review by fm3 — COMMENTED** [review]
Backend LGTM :) I added only the suggestion we already talked about, to use the enum values directly and not wrap strings.

Also, I assume you’ll update the result link for meshes as discussed, without the return value

File: app/views/mail/jobSuccessfulGeneric.scala.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      border: 1px solid #808cff;
+      text-align: center;
+      font-weight: 400;
+      font-size: 14px;
+      line-height: 2.1;
+      text-decoration: none;
+      border-radius: 4px;
+      display: inline-block;
+    ">
+    Click here to open
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply app/views/mail/jobSuccessfulGeneric.scala.html:24]: Done

File: app/views/mail/jobFailedUploadConvert.scala.html (hunk 2 above)
Comment (hotzenklotz) [reply app/views/mail/jobFailedUploadConvert.scala.html:7]: Done

**Comment by hotzenklotz** [comment]
@philippotto @fm3 I added your PR feedback

**Review by fm3 — APPROVED** [review]
Backend LGTM

**Review by philippotto — APPROVED** [review]

**Pull request merged**